File: ValueAnimatorExample/app/src/main/java/com/example/valueanimatorexample/MainActivity.java
package com.example.valueanimatorexample;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.os.Bundle;
import android.view.View;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.AnimationSet;
import android.view.animation.LinearInterpolator;
import android.widget.Button;
import android.widget.FrameLayout;

public class MainActivity extends AppCompatActivity {
    private Button buttonStart;
    private Button buttonEnd;
    private FrameLayout object;
    private ValueAnimator rotateY;
    private ValueAnimator alpha;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        buttonStart = findViewById(R.id.button_start);
        buttonEnd = findViewById(R.id.button_end);
        object = findViewById(R.id.object);

        setAnimation();
        setListener();
    }

    public void setListener(){
        buttonStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
            }
        });

        buttonEnd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
            }
        });
    }

    public void setAnimation(){
        rotateY = new ValueAnimator().ofFloat(0f, -180, 0f);
        rotateY.setDuration(2000);
        rotateY.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(@NonNull ValueAnimator animation) {
                float value = (float) animation.getAnimatedValue();
            }
        });
    }
}

File: LearnBasicXml/app/src/main/java/com/example/learnbasicxml/MainActivity.kt
package com.example.learnbasicxml

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import com.google.android.material.internal.NavigationMenu
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity() {

    private lateinit var nav_leftmenu : NavigationView;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // sử dụng màu cho menu
        nav_leftmenu = findViewById(R.id.nav_leftmenu);
        nav_leftmenu.itemIconTintList = null;
    }
}